### PR TITLE
Migrate to use AsyncHTTPClient

### LINKED ISSUE #4: Support AsyncHTTPClient
Hi! Thanks for the great package!

Have you considered adding support for using the [Swift Server Workgroup’s AsyncHTTPClient](https://github.com/swift-server/async-http-client/) for requests as an alternative to URLSession/URLRequest? That’s frequently more convenient/optimal in Swift on the server because it works nicely with SwiftNIO, where FoundationNetworking doesn’t.

Some examples:
- Vapor’s Client API is built with AsyncHTTPClient https://docs.vapor.codes/basics/client/
- All of Hummingbird’s exampels use AsyncHTTPClient https://github.com/hummingbird-project/hummingbird-examples/tree/main/proxy-server
- Soto (S3 SDK) is based on AsyncHTTPClient https://github.com/soto-project/soto
- etc.

For my use case it would make sense to just replace the implementation of `AppStoreServerAPIClient` entirely, but I understand that sometimes it makes sense to just use Foundation.

You can support both & even put NIO/AsyncHTTPClient support into a different target if you like. One possible solution would be to create an `HTTPBackend` protocol with a `sendRequest(...) async throws -> ...` requirement and create conforming structs like `AppStoreServerURLSessionHTTPBackend` and `AppStoreServerNIOHTTPBackend` that do the heavy lifting.

### PR DESCRIPTION
**Pull request by 0xTim**
URLSession is broken, incomplete and generally just a mess on Linux. For server-side applications to recommendation is to use [AsyncHTTPClient](https://github.com/swift-server/async-http-client). This PR migrates from `URLSession` to use AsyncHTTPClient instead.

Resolves #4

### CODE REVIEW

**Comment by alexanderjordanbaker** [comment]
@0xTim This seems to have a merge conflict with your other PR I merged a few days ago

**Comment by 0xTim** [comment]
@alexanderjordanbaker resolved!

**Comment by alexanderjordanbaker** [comment]
Hey @0xTim as you saw I'm messing around with the errors found in your build, do you have any ideas on those? I didn't get them building locally

**Comment by 0xTim** [comment]
@alexanderjordanbaker SwiftPM issue, think the latest build should fix it

File: Sources/AppStoreServerLibrary/ChainVerifier.swift [hunk 1]
```diff
@@ -150,31 +149,22 @@ final class AppStoreOIDPolicy: VerifierPolicy {
...
+                try? httpClient.syncShutdown()
+            }
+            var urlRequest = HTTPClientRequest(url: uri)
+            urlRequest.method = .POST
+            urlRequest.headers.add(name: "Content-Type", value: "application/ocsp-request")
+            urlRequest.body = .bytes(request)
+            let response = try await httpClient.execute(urlRequest, timeout: .seconds(30))
+            var body = try await response.body.collect(upTo: 1024 * 1024)
+            guard let data = body.readData(length: body.readableBytes) else {
+                return .nonTerminalError(OCSPFetchError())
```
Comment (alexanderjordanbaker) [inline Sources/AppStoreServerLibrary/ChainVerifier.swift:164]: Why is this not a terminalError? If we are unable to read data from the OCSP request, wouldn't that be an unrecoverable error?

Comment (0xTim) [reply Sources/AppStoreServerLibrary/ChainVerifier.swift:164]: Err I seem to remember it being caught elsewhere but yeah you're right. I'll update

**Review by alexanderjordanbaker — APPROVED** [review]
LGTM, thank you for this work

**Pull request merged**